lib/runners/processor/detekt.rb

```diff
@@ -0,0 +1,371 @@
+      )
+    end
+
+    def construct_result(report_id, report_path)
+      output_file_path = current_dir / report_path
+      if output_file_path.exist?
+        trace_writer.message "Reading output from #{report_path}..."
+        output = output_file_path.read
+      else
+        msg = "#{report_path} does not exist. Unexpected error occurred, processing cannot continue."
+        trace_writer.error msg
+        raise msg
+      end
+
+      switch_constructor(report_id, output)
+    end
+
+    def switch_constructor(report_id, output)
+      case report_id
+      when "xml"
+        construct_checkstyle_result(output)
+      when "txt"
+        construct_plain_result(output)
+      end
+    end
+
+    def construct_checkstyle_result(output)
+      document = REXML::Document.new(output)
+      unless document.root
+        msg = "Invalid XML output!"
+        trace_writer.error msg
+        raise msg
+      end
+
+      issues = []
+
+      document.root.each_element("file") do |file|
+        file.each_element do |error|
+          case error.name
+          when "error"
+            regexp = /.*\.(.*)/
+            rule, = error[:source].match(regexp).captures
+
+            issues << construct_issue(
+              file: file[:name],
+              line: error[:line],
+              message: error[:message],
+              rule: rule,
+            )
+          else
+            add_warning error.text.strip, file: file[:name]
+          end
+        end
+      end
+
+      issues
+    end
+
+    def construct_plain_result(output)
+      issues = []
+
+      regexp_a = /(.*) - .* - .* at (.*):(.*):(.*) - (.*)/
+      regexp_b = /(.*) - .* at (.*):(.*):(.*) - (.*)/
+      
+      output.lines(chomp: true).map do |line|
+        match = line.match(regexp_a)
+        if match.nil?
+          match = line.match(regexp_b)
+        end
+
+        unless match.nil?
+          rule, file, start_line, _, message = match.captures
+          issues << construct_issue(
+            file: file,
+            line: start_line,
+            message: message,
+            rule: rule,
+          )
+        end
+      end
+
+      issues
+    end
+
+    def construct_issue(file:, line:, message:, rule:)
+      Issue.new(
+        path: relative_path(working_dir.realpath / file, from: working_dir.realpath),
+        location: Location.new(start_line: line),
+        id: rule,
+        message: message,
+      )
+    end
+
+    def cli_baseline
+      cli_config[:baseline].then { |value| value ? ["--baseline", value] : [] }
+    end
+
+    def cli_config_files
+      cli_config[:config].then { |value| value.present? ? ["--config", value.join(",")] : [] }      
+    end
+
+    def cli_config_resource
+      cli_config[:"config-resource"].then { |value| value.present? ? ["--config-resource", value.join(",")] : [] }
+    end
+
+    def cli_disable_default_rulesets
+      "--disable-default-rulesets" if cli_config[:"disable-default-rulesets"]
```

Comment:
[suggestion] A small refactoring for consistency.

```suggestion
      cli_config[:"disable-default-rulesets"] ? ["--disable-default-rulesets"] : []
```